sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -239,25 +239,29 @@ public ConfigurationClientBuilder endpoint(String endpoint) {
      * @param connectionString Connection string in the format "endpoint={endpoint_value};id={id_value};
      * secret={secret_value}"
      * @return The updated ConfigurationClientBuilder object.
-     * @throws NullPointerException If {@code credential} is {@code null}.
+     * @throws NullPointerException If {@code connectionString} is {@code null}.
+     * @throws IllegalArgumentException if {@code connectionString} is an empty string. Or the secret is invalid and
+     * cannot instantiate HMAC-SHA256 MAC algorithm.
```

Comment:
Should this `@throws` statement be this?

```java
@throws IllegalArgumentException If {@code connectionString} is an empty string, the {@code connectionString} secret is invalid, or the HMAC-SHA256 MAC algorithm cannot be instantiated.
```

Mainly the second sentence is an issue since it isn't an and between an invalid secret and instantiating the algorithm but an or.